contentful/export-processor/content-types/section.js

```diff
@@ -152,6 +152,133 @@ export class Section {
+
+        // Create paths until all answers are used
+        while (answersUsed.length !== allAnswers.length) {
+            let newPath = [];
+            const startIndex = this.questions.findLastIndex(question => question.answers.some(answer => !answersUsed.includes(answer.id)));
+
+            if (startIndex === -1) {
+                console.log("All answers have been used.");
+                break;
+            }
+
+            // Find the latest question in the sequence that has unused answers and assign an unused answer
+            const lastQuestionWithUnusedAnswers = this.questions[startIndex];
+            const lastUnusedAnswers = lastQuestionWithUnusedAnswers.answers.filter(answer => !answersUsed.includes(answer.id));
+            let latestAnswer = this.assignBestAnswer(lastUnusedAnswers, startIndex);
+            let nextNeeded = lastQuestionWithUnusedAnswers.id;
+
+            newPath.unshift({ question: lastQuestionWithUnusedAnswers, answer: latestAnswer });
+
+            // Loop backwards through questions leading up to the last with unused answers, adding answers that lead to needed questions
+            for (let i = startIndex - 1; i >= 0; i--) {
+                const currentQuestion = this.questions[i];
+                let currentAnswer;
+                const unusedAnswers = currentQuestion.answers.filter(answer => !answersUsed.includes(answer.id));
+
+                // Find answers that lead to the next question in the sequence, prioritising unused answers and skipping any that jump ahead
+                if (unusedAnswers.some(answer => answer.nextQuestion?.sys.id === nextNeeded)) {
+                    currentAnswer = unusedAnswers.find(answer => answer.nextQuestion.sys.id === nextNeeded);
+                } else if (currentQuestion.answers.some(answer => answer.nextQuestion.sys.id === nextNeeded)) {
+                    currentAnswer = currentQuestion.answers.find(answer => answer.nextQuestion.sys.id === nextNeeded);
+                } else {
+                    console.log(`No valid answer found for question ${currentQuestion.id}. Skipping.`);
+                }
+
+                if (currentAnswer) {
+                    nextNeeded = currentQuestion.id;
+                    newPath.unshift({ question: currentQuestion, answer: currentAnswer });
+                }
+            }
+
+            // Loop forwards from the last question with unused answers, adding answers to complete the sequence, prioritising ending early or skipping questions
+            while (latestAnswer && latestAnswer.nextQuestion) {
+                const nextIndex = this.questions.findLastIndex(question => question.id === latestAnswer.nextQuestion.sys.id);
+                const nextQuestion = this.questions[nextIndex];
+                const nextAnswer = this.assignBestAnswer(nextQuestion.answers, nextIndex);
+                newPath.push({ question: nextQuestion, answer: nextAnswer });
+                latestAnswer = nextAnswer;
```

Comment:
🎉 v nice

Reply:
It feels very slippery in my brain and I'm still angry about it